src/main/java/org/datadog/jmxfetch/JmxAttribute.java

```diff
@@ -109,7 +112,14 @@ private void applyTagsBlackList() {
 
             for (String excludedTagName : include.getExcludeTags()) {
                 for (Iterator<String> it = this.defaultTagsList.iterator(); it.hasNext(); ) {
-                    if (it.next().startsWith(excludedTagName + ":")) {
+                    String tag = it.next();
+                    if (tag.startsWith("service:")) {
+                        if (tag.equals("service:" + this.serviceName)) {
+                            continue;
+                        } else {
+                            it.remove();
+                        }
```

Comment:
To explain in more details what I had in mind:

1. I'd make the `Instance` not add the service tag to the list of tags here: https://github.com/DataDog/jmxfetch/blob/a80913308d3c38e9b5abd63c2035df0161ca75a8/src/main/java/org/datadog/jmxfetch/Instance.java#L164-L166
2. then we can remove all this logic here, and simply add the service tag (if the `service` is not null or empty) to the list of tags after `applyTagsBlackList` is called here: https://github.com/DataDog/jmxfetch/blob/a80913308d3c38e9b5abd63c2035df0161ca75a8/src/main/java/org/datadog/jmxfetch/JmxAttribute.java#L475-L479 (and probably worth a comment explaining why we use this order of evaluation).

I think that it would make the logic of the service field handling easier to follow, but maybe wasn't worth all the time spent on this PR....